pkg/repositories/transformers/execution.go

```diff
@@ -228,6 +236,41 @@ func UpdateExecutionModelState(
 	return nil
 }
 
+// UpdateExecutionModelStateChangeDetails Updates an existing model with state and stateUpdateBy details from the request
+func UpdateExecutionModelStateChangeDetails(executionModel *models.Execution, state admin.ExecutionState,
+	stateUpdatedBy string) error {
+
+	var closure admin.ExecutionClosure
+	err := proto.Unmarshal(executionModel.Closure, &closure)
+	if err != nil {
+		return errors.NewFlyteAdminErrorf(codes.Internal, "Failed to unmarshal execution closure: %v", err)
+	}
+	// Update the indexed columns
+	stateInt := int32(state)
+	executionModel.State = &stateInt
+	occurredAt := time.Now()
```

Comment:
can we pass this in to the function and determine the request time in the pkg/rpc/ or manager layer?

Reply:
Done